search.go

```diff
@@ -0,0 +1,91 @@
+package main
+
+import (
+	"bytes"
+	"encoding/hex"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/tidwall/gjson"
+)
+
+// DefaultResponseBodyTitle defines the default header for the body view
+const DefaultResponseBodyTitle = "Response body (F9)"
+
+// Search performs a search on the request
+// returns title, display, error
+func Search(request *Request, searchString string) (string, []byte, error) {
+	isBinary := (strings.Index(request.ContentType, "text") == -1 &&
+		strings.Index(request.ContentType, "application") == -1)
+
+	if isBinary {
+		return DefaultResponseBodyTitle + " [binary content]",
+			[]byte(hex.Dump(request.RawResponseBody)), nil
+	}
+
+	if strings.Index(request.ContentType, "text") != -1 {
+		return SearchText(request, searchString)
+	}
+
+	if request.ContentType == "application/json" {
```

Comment:
Matching the whole ContentType string does not work here - for instance, a `Content-Type: application/json; charset=utf-8` results in `[unknown content]`, and neither gjson nor textual search works in this case.